src/AutoMapper/Mappers/FromStringDictionaryMapper.cs

```diff
@@ -31,6 +45,19 @@ private static object MapDynamic(StringDictionary source, object boxedDestinatio
                 MapInnerProperties();
             }
             return boxedDestination;
+            Match MatchSource(string name)
+            {
+                if (source.TryGetValue(name, out var value))
+                {
+                    return new Match { Value = value, Count = 1 };
+                }
+                var matches = source.Where(s => s.Key.Trim() == name).Select(s=>s.Value).ToArray();
```

Comment:
I guess this is inefficient by design :) Hopefully nobody will pass a huge dictionary here.

Reply:
Maybe take a moment to consider whether we really want this :)